Introduction of Yoda condition checking.

dirtydanee opened the pull request:
This PR aims to implement [#1924][1].

[1]: https://github.com/realm/SwiftLint/issues/1924

coverback commented:
This is amazing!

But there are some false positives, like this one: https://github.com/mozilla-mobile/firefox-ios/blob/61d2f9e8d8c8de169c564642563d7c2c97f456a8/Client/Frontend/Settings/SearchSettingsTableViewController.swift#L249:9

coverback commented:
Have you thought about having `nil` here as well? Or should I ask about it in the #1924?

dirtydanee commented:
Hey @coverback, glad you like it!

I have fixed the false positives, and i agree, the `nil` check is a good idea! I have added support for that case too.
